akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -0,0 +1,115 @@
+import scala.compat.java8.FunctionConverters._
+
+@ApiMayChange
+object LfuCache {
+
+  def apply[K, V](implicit system: ActorSystem): akka.http.caching.scaladsl.Cache[K, V] =
+    apply(akka.http.caching.scaladsl.CachingSettings(system))
+
+  /**
+   * Creates a new [[akka.http.caching.LfuCache]], with optional expiration depending
+   * on whether a non-zero and finite timeToLive and/or timeToIdle is set or not.
+   */
+  def apply[K, V](cachingSettings: CachingSettings): akka.http.caching.scaladsl.Cache[K, V] = {
+    val settings = cachingSettings.lfuCacheSettings.asInstanceOf[LfuCacheSettings]
+
+    require(settings.maxCapacity >= 0, "maxCapacity must not be negative")
+    require(settings.initialCapacity <= settings.maxCapacity, "initialCapacity must be <= maxCapacity")
+
+    if (settings.timeToLive.isFinite() || settings.timeToIdle.isFinite()) expiringLfuCache(settings.maxCapacity, settings.initialCapacity, settings.timeToLive, settings.timeToIdle)
+    else simpleLfuCache(settings.maxCapacity, settings.initialCapacity)
+  }
+
+  /**
+   * Java API
+   * Creates a new [[akka.http.caching.LfuCache]] using configuration of the system,
+   * with optional expiration depending on whether a non-zero and finite timeToLive and/or timeToIdle is set or not.
+   */
+  def create[K, V](system: ActorSystem): akka.http.caching.javadsl.Cache[K, V] =
+    apply(system)
+
+  /**
+   * Java API
+   * Creates a new [[akka.http.caching.LfuCache]], with optional expiration depending
+   * on whether a non-zero and finite timeToLive and/or timeToIdle is set or not.
+   */
+  def create[K, V](settings: CachingSettings): akka.http.caching.javadsl.Cache[K, V] =
+    apply(settings)
+
+  private def simpleLfuCache[K, V](maxCapacity: Int, initialCapacity: Int): LfuCache[K, V] = {
+    val store = Caffeine.newBuilder().asInstanceOf[Caffeine[K, V]]
+      .initialCapacity(initialCapacity)
+      .maximumSize(maxCapacity)
+      .buildAsync[K, V](dummyLoader[K, V])
+    new LfuCache[K, V](store)
+  }
+
+  private def expiringLfuCache[K, V](maxCapacity: Long, initialCapacity: Int,
+                                     timeToLive: Duration, timeToIdle: Duration): LfuCache[K, V] = {
+    require(
+      !timeToLive.isFinite || !timeToIdle.isFinite || timeToLive > timeToIdle,
+      s"timeToLive($timeToLive) must be greater than timeToIdle($timeToIdle)")
+
+    def ttl: Caffeine[K, V] ⇒ Caffeine[K, V] = { builder ⇒
+      if (timeToLive.isFinite) builder.expireAfterWrite(timeToLive.toMillis, TimeUnit.MILLISECONDS)
+      else builder
+    }
+
+    def tti: Caffeine[K, V] ⇒ Caffeine[K, V] = { builder ⇒
+      if (timeToIdle.isFinite) builder.expireAfterAccess(timeToIdle.toMillis, TimeUnit.MILLISECONDS)
+      else builder
+    }
+
+    val builder = Caffeine.newBuilder().asInstanceOf[Caffeine[K, V]]
+      .initialCapacity(initialCapacity)
+      .maximumSize(maxCapacity)
+
+    val store = (ttl andThen tti)(builder).buildAsync[K, V](dummyLoader[K, V])
```

Comment:
It should be possible to get rid of `simpleLfuCache` altogether since `expiringLfuCache` "degrades" based on the durations.